File: Easy/653_Two_sum_IV_input_is_a_BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int>intarray;
    void checkarray(TreeNode* root)
    {
        if(root)
        {
            checkarray(root->left);
            intarray.push_back(root->val);
            checkarray(root->right);
        }
    }
    bool findTarget(TreeNode* root, int k) {
        checkarray(root);
        int i=0,j=intarray.size()-1;
        while(i<j)
        {
            if(intarray[i]+intarray[j]==k)
            return true;
            if(intarray[i]+intarray[j]<k)
            {
                i++;
            }
            if(intarray[i]+intarray[j]>k)
            {
                j--;
            }
        }
        return false;

    }
};

File: Easy_python/653_two_sum_IV-Input_is_a_BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def findTarget(self, root: TreeNode, k: int) -> bool:
        res=[]
        if root:
            def dfs(root):
                if root:
                    
                    dfs(root.left)
                    res.append(root.val)
                    dfs(root.right)
            dfs(root)
        left=0
        right=len(res)-1
        while left<right:
            if res[left]+res[right]==k:
                return True
            elif res[left]+res[right]<k:
                left=left+1
            else:
                right=right-1
        return False
